R/raincloudPlots.R

```diff
@@ -0,0 +1,908 @@
+#
+# Copyright (C) 2013-2024 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+
+# Main function: raincloudPlots() ----
+raincloudPlotsInternal <- function(jaspResults, dataset, options) {
+  ready    <- (length(options$dependentVariables) > 0)
+  dataset  <- .rainReadData(dataset, options)
+  dataInfo <- .rainDataInfo(dataset, options)
+  .rainCreatePlots( jaspResults, dataInfo, options, ready)
+  .rainCreateTables(jaspResults, dataInfo, options, ready)
+}  # End raincloudPlots()
+
+
+
+# .rainReadData() ----
+.rainReadData <- function(dataset, options) {
+
+  if(!is.null(dataset)) {
+    output <- dataset
+  } else {
+
+    # Step 1: Read in all variables that are given by JASP; if no input, then nothing is read in
+    columnsVector <- c(options$dependentVariables)
+    optionsVector <- c(options$primaryFactor, options$secondaryFactor, options$covariate, options$observationId)
+    for (option in optionsVector) {
+      if (option != "") columnsVector <- c(columnsVector, option)
+    }
+    datasetInProgress <- .readDataSetToEnd(columns = columnsVector)
+
+    # Step 2: Create columns with consistent names; if no input then assign default
+    datasetInProgress$primaryFactor   <- .rainDataColumn(datasetInProgress,  options$primaryFactor)
```

Comment:
options (and ideally all lists) should be accessed via [[""]] notation rather than "$"
i.e., ```options[["primaryFactor"]]```

please change throughout the analysis
(also see R guide)

Reply:
I would like to keep it as is for three reasons:

First, the JASP r-analyses-guide
https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/development/r-analyses-guide.md#step-3---reading-the-dataset
also suggests the $ operator:
![Screenshot 2024-04-15 at 16 31 35](https://github.com/jasp-stats/jaspDescriptives/assets/123652963/4c88941e-4d0a-4a52-8b81-d267fca7b6ad)

Second, Don never brought it up, so I thought it was fine as is.

Third, I think the code is tidier when using a single `$` compared to double brackets + quotes `[[""]]`.
(Unless in those cases where I am accessing the containers.)

Why do you suggest the change?

Reply:
I implemented it for the options.
Thus, now everything should be resolved and the rainclouds can go into the next nightly, right?